### PR TITLE
1151 shell to specific instance - part 1 - list application instances

### PR DESCRIPTION
**Pull request by jimmykarily**
This is a preparation for #1151

We first need a way to list application instances. Then the user can request a shell to a specific instance.
This PR changes the endpoint for `epinio app show`, to list the application instances with their metrics.

### CODE REVIEW

File: internal/application/workload.go [hunk 1]
```diff
@@ -469,3 +430,83 @@ func (a *Workload) Get(ctx context.Context) (*models.AppDeployment, error) {
...
+			Ready:     podutils.IsPodReady(&pods[i]),
+			CreatedAt: pod.ObjectMeta.CreationTimestamp.Time.Format(time.RFC3339), // ISO 8601
+		}
+	}
+
+	return result
+}
+
+func (a *Workload) populatePodMetrics(podInfos map[string]*models.PodInfo, podMetrics []metricsv1beta1.PodMetrics) error {
+	// Calculate metrics
```
Comment (jimmykarily) [inline internal/application/workload.go:484]: Unnecessary comment, will remove
```suggestion
```

File: acceptance/api/v1/application_show_test.go [hunk 2]
```diff
@@ -40,11 +41,17 @@ var _ = Describe("AppShow Endpoint", func() {
...
 		Expect(createdAt.Unix()).To(BeNumerically("<", time.Now().Unix()))
 
-		Expect(appObj.Workload.Restarts).To(BeNumerically("==", 0))
-
 		Expect(appObj.Workload.DesiredReplicas).To(BeNumerically("==", 1))
 		Expect(appObj.Workload.ReadyReplicas).To(BeNumerically("==", 1))
 
+		Expect(len(appObj.Workload.Replicas)).To(Equal(1))
+		var replica *models.PodInfo
+		for _, v := range appObj.Workload.Replicas {
```
Comment (jimmykarily) [inline acceptance/api/v1/application_show_test.go:49]: I realize now that we use the terms `Replica`, `Pod` and `Instance` in various places and we refer to the same thing. I will change it to be `Instance` everywhere (as in "application instance")

**Review by andreas-kupries — CHANGES_REQUESTED** [review]
Looks ok to me, although the test `lists the application data [It]` got broken. Apparently the closing `Eventually` does not happen.

**Comment by jimmykarily** [comment]
> Looks ok to me, although the test `lists the application data [It]` got broken. Apparently the closing `Eventually` does not happen.

I'll run that test locally to ensure it's passing. If yes, I will increase the 4s timeout to 15s. Maybe the "Restarts" field didn't have enough time to update itself?

**Comment by jimmykarily** [comment]
> > Looks ok to me, although the test `lists the application data [It]` got broken. Apparently the closing `Eventually` does not happen.
> 
> I'll run that test locally to ensure it's passing. If yes, I will increase the 4s timeout to 15s. Maybe the "Restarts" field didn't have enough time to update itself?

Test passes locally, I increased the timeout and rebased the branch to origin/main

**Review by andreas-kupries — APPROVED** [review]

**Pull request merged**